src/gen_swift/templates/module.swift

```diff
@@ -1,55 +1,102 @@
-class BreezSDKModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
-    private lateinit var executor: ExecutorService
-    private var breezServices: BlockingBreezServices? = null
-
-    companion object {
-        const val TAG = "RNBreezSDK"
-        const val GENERIC_CODE = "Generic"
+@objc(RNBreezSDK)
+class RNBreezSDK: RCTEventEmitter {
+    static let TAG: String = "BreezSDK"
+    
+    private var breezServices: BlockingBreezServices!
+    
+    static var breezSdkDirectory: URL {
+      let applicationDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
+      let breezSdkDirectory = applicationDirectory.appendingPathComponent("breezSdk", isDirectory: true)
+      
+      if !FileManager.default.fileExists(atPath: breezSdkDirectory.path) {
+        try! FileManager.default.createDirectory(atPath: breezSdkDirectory.path, withIntermediateDirectories: true)
+      }
+      
+      return breezSdkDirectory
     }
-
-    override fun initialize() {
-        super.initialize()
-
-        executor = Executors.newFixedThreadPool(3)
+    
+    @objc
+    override static func moduleName() -> String! {
+        TAG
     }
-
-    override fun getName(): String {
-        return TAG
+    
+    override func supportedEvents() -> [String]! {
+        return [BreezSDKListener.emitterName, BreezSDKLogStream.emitterName]
     }
-
-    @Throws(SdkException::class)
-    fun getBreezServices(): BlockingBreezServices {
-        if (breezServices != null) {
-            return breezServices!!
+    
+    @objc
+    override static func requiresMainQueueSetup() -> Bool {
+        return false
+    }
+    
+    func getBreezServices() throws -> BlockingBreezServices {
+        if breezServices != nil {
+            return breezServices
         }
-
-        throw SdkException.Generic("BreezServices not initialized")
+        
+        throw SdkError.Generic(message: "BreezServices not initialized")
     }
 
-    @ReactMethod
-    fun addListener(eventName: String) {}
-
-    @ReactMethod
-    fun removeListeners(count: Int) {}
+    {% let obj_interface = "BreezSDK." -%}
+    {% for func in ci.function_definitions() %}
+    {%- if func.name()|ignored_function == false -%}
+    {% include "TopLevelFunctionTemplate.swift" %}
+    {% endif -%}
+    {%- endfor %}  
+    @objc(defaultConfig:apiKey:nodeConfig:resolve:reject:)
+    func defaultConfig(_ envType: String, apiKey: String, nodeConfig: [String: Any], resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
+        do {
+            let envTypeTmp = try BreezSDKMapper.asEnvironmentType(type: envType)
+            let nodeConfigTmp = try BreezSDKMapper.asNodeConfig(data: nodeConfig)
+            var config = BreezSDK.defaultConfig(envType: envTypeTmp, apiKey: apiKey, nodeConfig: nodeConfigTmp)
+            config.workingDir = RNBreezSDK.breezSdkDirectory.path                
```

Comment:
The defaultConfig is not generated because of this line addition of setting the workingDir?
If this is the case maybe we can add an exception in the TopLevelFunctionTemplate and hook this line in case we are generating this method. Just looking to avoid having the prototype/mapping generated.

Reply:
Exactly, because of pre-setting the workingDir. I can also manage this in the TopLevelFunctionTemplate 👍

Reply:
Done